File: front-end/src/application/service/scaleService.ts
import { ScaleRepository } from "../repository/scaleRepository";
import { Scale } from "../../domain/entity/scale";
import { Either } from "../../utils/ull-either";
import { DataError } from "../../utils/ull-data-error";
import { UllUUID } from '../../utils/ull-uuid';

export class ScaleService {
    private scaleRepository: ScaleRepository;

    constructor(scaleRepository: ScaleRepository) {
        this.scaleRepository = scaleRepository;
    }
    
    public save = (newScale: Scale): Promise<Either<DataError, Scale>> => {
        return this.scaleRepository.save(newScale);
    }

    public find = (id: UllUUID): Promise<Either<DataError, Scale>> => {
        return this.scaleRepository.find(id);
    }

    public findAll = (pageRequest: any): Promise<Either<DataError, Scale[]>> => {
        return this.scaleRepository.findAll(pageRequest);
    }

    public deleteAll = (): Promise<Either<DataError, Scale[]>> => {
        return this.scaleRepository.deleteAll();
    }
}

File: back-end/src/main/java/com/trucks/application/service/ScaleService.java
package com.trucks.application.service;

import java.util.List;
import java.util.UUID;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageRequest;

import com.trucks.application.port.ScaleRepository;
import com.trucks.domain.entity.Scale;

public class ScaleService {

    @Autowired
    ScaleRepository scaleRepository;

    public Page<Scale> findAll(PageRequest page) {
        return this.scaleRepository.findAll(page);
    }

    public Optional<Scale> find(UUID id) {
        return this.scaleRepository.find(id);
    }

    public Scale save(Scale newScale) {
        return this.scaleRepository.save(newScale);
    }

    public List<Scale> deleteAll() {
        return this.scaleRepository.deleteAll();
    }
}
